crates/tests-e2e/tests/rollover_position.rs

```diff
@@ -0,0 +1,39 @@
+use native::api;
+use native::trade::position;
+use position::PositionState;
+use tests_e2e::setup;
+use tests_e2e::wait_until;
+
+#[tokio::test]
+#[ignore]
+async fn can_rollover_position() {
+    let test = setup::TestSetup::new_with_open_position().await;
+    let coordinator = &test.coordinator;
+    let dlc_channels = coordinator.get_dlc_channels().await.unwrap();
+    let app_pubkey = api::get_node_id().0;
+
+    tracing::info!("{:?}", dlc_channels);
+
+    let dlc_channel = dlc_channels
+        .into_iter()
+        .find(|chan| chan.counter_party == app_pubkey)
+        .unwrap();
+
+    coordinator
+        .rollover(&dlc_channel.dlc_channel_id.unwrap())
+        .await
+        .unwrap();
+
+    wait_until!(test
+        .app
+        .rx
+        .position()
+        .map(|p| PositionState::Rollover == p.position_state)
+        .unwrap_or(false));
+    wait_until!(test
+        .app
+        .rx
+        .position()
+        .map(|p| PositionState::Open == p.position_state)
+        .unwrap_or(false));
```

Comment:
🔧 I think we need a few more assertions to ensure that rollover has actually taken place. In theory we could got to the rollover state and back to the open state because rollover failed?

Reply:
> 🔧 I think we need a few more assertions to ensure that rollover has actually taken place. 
I've added an assertion on the extended expiry, what else would you suggest to assert on?

> In theory we could got to the rollover state and back to the open state because rollover failed?
Only if we implement it, but currently there is no flow from `Rollover` to `Open` if the the renew protocol fails. That reminds me that we need to consider the renew protocol as well for our timeout handling. (see also https://github.com/get10101/10101/issues/1108)